governance/program/src/state/legacy.rs

```diff
@@ -125,14 +125,6 @@ pub struct GovernanceV1 {
 
     /// Governance config
     pub config: GovernanceConfig,
-
-    /// Reserved space for future versions
-    pub reserved: [u8; 6],
-
-    /// The number of proposals in voting state in the Governance
-    /// Note: This is field introduced in V2 but it took space from reserved
-    /// and we have preserve it for V1 serialization roundtrip
-    pub voting_proposal_count: u16,
```

Comment:
Why are you removing the reserve entirely too?

Reply:
It was a bug that it was still there because all the reserved space was moved to `GovernanceConfig` in previous PRs for `V3`. It escaped me before and this is why I added the account size test as a guard https://github.com/solana-labs/solana-program-library/blob/e0dd05fa5caf929e8910346ac4bb6a0841ddbfdf/governance/program/src/state/governance.rs#L557

I should find a better way to test for those legacy accounts translations, but it might not be needed any longer because now we can migrate all the accounts on the fly to the latest version using `reallocs`

Reply:
it would be great to have basic unit tests for v1->v2 transitions for all concerned account types